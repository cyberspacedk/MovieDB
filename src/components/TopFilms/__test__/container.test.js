/* eslint-disable import/named */
/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import TopFilmsContainerConnected, {
  mstp,
  TopFilmsContainer,
} from '../container';
/*
 1- snapshot
 2- render one component

*/
describe('TopFilmsContainer ', () => {
  const mockStore = configureStore();
  const mockClick = jest.fn();
  let store;
  let wrapper;

  const initialState = {
    topFilms: {
      loading: false,
      films: [{}],
      error: false,
    },
  };
  const props = {
    fetchDataRequest: jest.fn(),
    loading: false,
    error: false,
  };

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<TopFilmsContainerConnected store={store} />);
  });

  it('Snapshot. Should Match', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render container component', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Shoud show previous loading value', () => {
    expect(mstp(initialState).loading).toEqual(false);
  });

  it('calls ComponentDidMount', () => {
    const wrapperDump = shallow(<TopFilmsContainer {...props} />);
    wrapperDump.instance().componentDidMount();
    expect(wrapperDump.instance().fetchDataRequest).toHaveBeenCalledTimes(1);
  });
});
