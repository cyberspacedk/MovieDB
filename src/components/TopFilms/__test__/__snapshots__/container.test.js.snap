// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFilmsContainer  Snapshot. Should Match 1`] = `
<ContextProvider
  value={null}
>
  <TopFilmsContainer
    error={false}
    fetchDataRequest={[Function]}
    loading={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    topFilms={
      Array [
        Object {},
      ]
    }
  />
</ContextProvider>
`;
